cookbooks/convection_box_3d/doc/convection_box_3d.md

```diff
@@ -155,32 +156,25 @@ approximately $t=0.01$ corresponds to a relative lull: some plumes rise up,
 but not very fast because the medium is now stably layered but not fully
 mixed. This can be seen in the relatively low heat fluxes, but also in the
 fact that there are almost horizontal temperature isosurfaces in the second of
-the pictures in Fig.&nbsp;[6][]. After that, the general structure of the
+the pictures in {numref}`fig:box-3d-solution`. After that, the general structure of the
 temperature field is that the interior of the domain is well mixed with a
 mostly constant average temperature and thin thermal boundary layers at the
 top and bottom from which plumes rise and sink. In this regime, the average
 heat flux is larger but also more variable depending on the number of plumes
 currently active. Many other analyses would be possible by using what is in
 the statistics file or by enabling additional postprocessors.
 
-<figure>
-<img src="cookbooks/convection_box_3d/doc/heat-flux.png" id="fig:box-3d-heat-flux" style="width:60.0%" alt="Convection in a 3d box: Upward heat flux through the bottom and top boundaries as a function of time." /><figcaption aria-hidden="true"><em>Convection in a 3d box: Upward heat flux through the bottom and top boundaries as a function of time.</em></figcaption>
-</figure>
+```{figure-md} fig:box-3d-heat-flux
+<img src="heat-flux.*" style="width:60.0%"/>
+
+Convection in a 3d box: Upward heat flux through the bottom and top boundaries as a function of time.
+```
 
-[1] For computations of this size, one should test a few time steps in debug
+[^footnote1]: For computations of this size, one should test a few time steps in debug
 mode but then, of course, switch to running the actual computation in
-optimized mode &ndash; see Section&nbsp;[\[sec:debug-mode\]][4].
+optimized mode &ndash; see Section&nbsp;{ref}`sec:debug-mode`.
 
-[2] Note that the statistics file actually contains the *outward* heat flux
+[^footnote2]: Note that the statistics file actually contains the *outward* heat flux
 for each of the six boundaries, which corresponds to the *negative* of upward
 flux for the bottom boundary. The figure therefore shows the negative of the
-values available in the statistics file.
-
-  [cookbooks/convection_box_3d.prm]: cookbooks/convection_box_3d.prm
-  [1]: #parameters:Geometry_20model
-  [2]: #parameters:global
-  [6]: #fig:box-3d-solution
-  [9]: #fig:box-3d-mesh
-  [3]: #fig:convection-box-fields
-  [10]: #fig:box-3d-heat-flux
-  [4]: #sec:debug-mode
+values available in the statistics file.
```

Comment:
Our indentation tester requires a final newline character at the end of each file. This file is currently missing one.
```suggestion
values available in the statistics file.

```